File: farewell.py
#!/usr/bin/env python

import sys, re

key = "i hope in the next ten years there would be no other farewell letter brilliant than this one"

def decrypt(content):
    encrypted = re.sub("\s", "", content).split(",")
    decrypted = "".join([chr(int(encrypted[i]) ^ ord(key[i % len(key)])) for i in range(len(encrypted))])
    return decrypted

with open(sys.argv[1], "r") as f:
    print decrypt(f.read()),

File: farewell.pl
#!/usr/bin/env perl

my $key = "i hope in the next ten years there would be no other farewell letter brilliant than this one";

open FILE, "$ARGV[0]" or die $!;
my ($content, $buffer, $n);
while (($n = read FILE, $buffer, 32) != 0) {
  $content .= $buffer;
}
close(FILE);

$content =~ s/\s//g;
my @encrypted = split(/,/, $content);
my $length = @encrypted;

my @decrypted;
for (my $i = 0; $i < $length; ++$i) {
    my $tmp = int(@encrypted[$i]) ^ ord(substr($key, $i % length($key), 1));
    push(@decrypted, chr($tmp));
}

print @decrypted;
